jsonpatcher/jsonpatcher.go

```diff
@@ -134,12 +142,22 @@ func (jp *jsonPatcher) Reduce(events []core.Event, datastore ds.TxnDatastore, ba
 			if err = txn.Put(key, patchedValue); err != nil {
 				return nil, err
 			}
+			if err := indexFunc(e.Model(), key, value, patchedValue, txn); err != nil {
+				return nil, fmt.Errorf("error when indexing created data: %v", err)
+			}
 			actions[i] = core.ReduceAction{Type: core.Save, Model: e.Model(), EntityID: e.EntityID()}
 			log.Debug("\tsave operation applied")
 		case delete:
+			value, err := txn.Get(key)
```

Comment:
Because we use the values to generate the index, to remove the index, we need the original value again...

Reply:
This is a big piece of the indexing overhead.